pkg/hs/king/lib/Ur/King/Main.hs

```diff
@@ -185,8 +189,8 @@ runOrExitImmediately getPier oExit =
 tryPlayShip :: ( HasLogFunc e, HasNetworkConfig e, HasPierConfig e
                , HasConfigDir e
                )
-            => Bool -> Bool -> Serf.Flags -> RIO e ()
-tryPlayShip exitImmediately fullReplay flags = do
+            => Bool -> Bool -> Maybe Word64 -> Serf.Flags -> RIO e ()
+tryPlayShip exitImmediately fullReplay playFrom flags = do
```

Comment:
This is bad code. If you look at the type of this function, you have no idea what the arguments are for. Generally, the solution to this is to create a record or to use newtypes to make the types more descriptive.

Again, not your fault. This was already bad, just making style notes since you asked for them.